docs/api/net.md

```diff
@@ -168,6 +171,9 @@ This method will resolve hosts from the [default
 session](session.md#sessiondefaultsession). To resolve a host from
 another session, use [ses.resolveHost()](session.md#sesresolvehosthost-options).
 
+Note: `net.resolveHost` is not available in the
+[utility process](../glossary.md#utility-process).
```

Comment:
I think this is fixable; we can call `ResolveHost` using the system network context.

... 🤔 This also makes me wonder if all the `net.*` methods should use the system network context instead of redirecting to `session.defaultSession`. It would be a breaking change but possibly a good idea? As is, there's no way to do net requests outside of a session context in the main process. For consideration for a followup, in any case.

Reply:
I think I almost got this working. Here's my current work in progress https://github.com/electron/electron/commit/df6da0ecc17f1e0f96de21701746a8609a02923b so far all it's profited me is a -2 generic `net::ERR_FAILED`. If anyone can spot what I'm doing wrong or knows how to debug further I would be very grateful 🙏 

The tricky bit is calling the network context from the utility process. I'm using the [NetworkContext.CreateHostResolver](https://source.chromium.org/chromium/chromium/src/+/main:services/network/public/mojom/network_context.mojom;bpv=1;bpt=1;l=1493?q=%22HostResolver%22%20f:mojom$&ss=chromium%2Fchromium%2Fsrc&gsn=CreateHostResolver&gs=KYTHE%3A%2F%2Fkythe%3A%2F%2Fchromium.googlesource.com%2Fcodesearch%2Fchromium%2Fsrc%2F%2Fmain%3Flang%3Dmojom%23network.mojom.NetworkContext.CreateHostResolver) to create a `pending_remote` that can be passed to the utility process.

Another approach could be to pass the whole `NetworkContext` but I'm not sure if that would be desirable (and also I'm not sure how to do it 😏)

Reply:
Ok fixed this in cc4fb8f I was just mishandling the `Remote` reference 🙃

Reply:
For the question of using the system network context over the default session, I'm curious if these are functionally equivalent? @deepak1556 I believe you mentioned something to that effect, if I understood correctly.

Reply:
hm I just tested this out by setting a proxy with `session.defaultSession.setProxy` in the main process and it doesn't appear that the system network context from the utility process gets the setting. @deepak1556 did I misunderstand what you mentioned or is there something VS Code is doing additionally?

Reply:
The system network context and the default session context are different. The system context is for network requests issued by "the system", e.g. when Chromium itself makes a network request for whatever reason, whereas the default session context is the context from which network requests made by web contents in the default session come.

Reply:
> This also makes me wonder if all the net.* methods should use the system network context instead of redirecting to session.defaultSession. It would be a breaking change but possibly a good idea? As is, there's no way to do net requests outside of a session context in the main process. For consideration for a followup, in any case.

I like this idea in which case we would also need to provide a way to configure proxy `net.setProxy`, I can tackle this as a follow-up.